### PR TITLE
consolidate sparse info into table classes

### LINKED ISSUE #20174: Refactor reviewer queues to gather title, permission, url in one place
The various properties of reviewer queues are scattered across multiple files: `urls.py`, `views.py`, `jinja_helpers.py`, `utils.py`, `queue.html`. We should consolidate all that. Maybe add `title`, `permission` and `urlname` to the various `Table` classes, add a registry `dict` gathering all queues, and then use that everywhere.

The registry already exists, `reviewer_tables_registry`. So really it's all about putting more stuff in the table classes and using that.

### PR DESCRIPTION
**Pull request by GregSherman**
Fixes #20174 

Move much of the information of reviewer queues into their table classes. Remove the need for alias queue functions.

### CODE REVIEW

File: src/olympia/reviewers/urls.py [hunk 1]
```diff
@@ -13,31 +13,43 @@
         r'^dashboard$', lambda request: redirect('reviewers.dashboard', permanent=True)
     ),
     re_path(
-        r'^queue/extension$', views.queue_extension, name='reviewers.queue_extension'
+        getattr(views.reviewer_tables_registry['extension'], 'url'),
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/reviewers/urls.py:16]: Btw, the next step with these urls is to replace them with a single `re_path(r'^queue/', include(<something that creates an individual re_path for each item in reviewer_tables_registry>))`

**Review by eviljeff — COMMENTED** [review]
You've got an import error in your tests that is likely hiding some fails.

**Comment by eviljeff** [comment]
> * I can't seem to update anything in jinja_helpers as any import from views or utils breaks the build.

Probably from a circular reference error?  The fix it usually to pass whatever you need into the template as a kwarg, and then pass it on to the jinja_helper function.  Note the permission checking in `queue_tabnav` might need leaving to a follow-ip.

> * I use the registry for both the name and url in urls.py when it may be appropriate to do just the name there.

I think using it for as much as possible is what we're aiming for.

> * The moderation queue, for consistency's sake, should there be a pointer in the registry to some object containing all of the values required for consolidation?

Yeah... but at this point I'm leaning towards touching the moderation queue in a follow-up pull request.  I can think of a few ways to do it but it'd be ideal to get everything else working.

File: src/olympia/reviewers/views.py [hunk 2]
```diff
@@ -298,37 +298,44 @@ def save_motd(request):
     data = context(form=form)
     return TemplateResponse(request, 'reviewers/motd.html', context=data)
 
+def queue(request, tab):
+    @permission_or_tools_listed_view_required(getattr(reviewer_tables_registry[tab], 'permission', None))
+    def _queue(request, tab, unlisted=False):
```
[the remark this answers is not in the document]
Comment (GregSherman) [reply src/olympia/reviewers/views.py:303]: I think the unlisted queue was removed a while ago and this is just remnant code. Had to do the same in queue.html

File: src/olympia/reviewers/urls.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (GregSherman) [reply src/olympia/reviewers/urls.py]: Would this entail having some sort of "make_queue_urls" function directly within the urls.py and calling it inside urlpatterns?

**Comment by GregSherman** [comment]
> Note the permission checking in `queue_tabnav` might need leaving to a follow-ip.

> Yeah... but at this point I'm leaning towards touching the moderation queue in a follow-up pull request.

In terms of the tabnav builder, should they be condensed into a loop with a special case for the moderation queue (little messy), or should that wait on the handling of the moderation queue special case in the follow-up pull request?

**Comment by eviljeff** [comment]
> > Note the permission checking in `queue_tabnav` might need leaving to a follow-ip.
> 
> > Yeah... but at this point I'm leaning towards touching the moderation queue in a follow-up pull request.
> 
> In terms of the tabnav builder, should they be condensed into a loop with a special case for the moderation queue (little messy), or should that wait on the handling of the moderation queue special case in the follow-up pull request?

Lets go for special case for moderation queue, unless it gets too messy and you find yourself having to write a lot of code that'll be thrown away again.

**Pull request merged**